### DISCUSSION TITLE
Option to disable checking for timestamps

### DISCUSSION THREAD

**Discussion by dploeger** [abraunegg/onedrive#3013] [Q&A] [answered]
Hey!

We're using the onedrive client to sync an export of a confluence space to our business onedrive. As it is an export that is generated every night, the timestamps of the files differ obviously. Now, onedrive seems to think that these files have actually changed and uploads nearly all of them every night (I wonder why only nearly all of them and not ALL of them).

I guess that is because of the different timestamps. Can this be disabled so that the sync only checks for actual changes in the content?

Thanks in advance

Kind regards
Dennis

**Comment by abraunegg** [comment, marked as answer]
@dploeger 

> I guess that is because of the different timestamps. Can this be disabled so that the sync only checks for actual changes in the content?

Please review the client architecture documentation: https://github.com/abraunegg/onedrive/blob/master/docs/client-architecture.md

The client does 2 passes:
1. timestamp
2. hash

The timestamp is used first, as this is computationally less intensive.

If the timestamp is different, the hash is checked. If the hash is indeed different, action is taken.

If the has is the same, the offending timestamp is attempted to be corrected and zero transfer is done.

**Reference:** https://github.com/abraunegg/onedrive/blob/master/docs/client-architecture.md#determining-if-an-item-is-synchronised-between-microsoft-onedrive-and-the-local-file-system

Please also ensure you are using v2.5.3 or above as there are *many* fixes for file comparison between v2.4.25 and v2.5.x codebase.

**Reply by dploeger** [reply]
Thanks. We have switched back to the latest tag, because edge was too instable. Does the latest tag already include 2.5.3?

**Reply by abraunegg** [reply]
Please clarify what you mean by 'edge' being unstable?

**Reply by dploeger** [reply]
Our job runs every night and on edge we had some errors in the synchronization workflow. Do you consider edge to be stable? I thought it would be the bleeding edge of development where bugs could happen. That's why I switched to latest, which I thought would be more stable.

**Reply by abraunegg** [reply]
I would consider 'edge' to be more stable given this is where bugfixes for bugs in 'latest' have been fixed. Fixes so far in 'edge' are:
* Fix build error on OpenBSD - technically irrelevant for a Docker image
* Fix unable to parse OpenSSL version - impacts Docker images
* Fix importation of 'sync_list' rules - impacts Docker images
* Fix --display-config when dealing with empty sync_list + ignore empty lines - impacts Docker images
* Fix that Business Shared Items shortcuts are skipped - impacts Docker images
* Fix regression that sync_list does not traverse shared directories  - impacts Docker images
* Add quota status messages based on OneDrive Account API response - not really a bug fix, but additional helpful application output
* Implement Feature Request: Support Permanent Delete on OneDrive - not a bug fix, but this can be applied to Docker containers if required
* Check for no --sync | --monitor earlier to fail fast if not used - not a bug fix, but application startup flow improvement

If you are having *issues* on 'edge' I would be raising a bug report.

**Reply by dploeger** [reply]
Ah okay. Then I'll switch back to edge. Thanks.
